File: app/internal/publisher/template/go/widgets/status_widget.go
package widgets

import (
	"fmt"

	"github.com/websmith/sdk"
)

type Status struct {
	engine  *wsd.LicenseEngine
	label   string
	color   string
}

func NewStatus(engine *wsd.LicenseEngine) *Status {
	return &Status{
		engine: engine,
	}
}

func (s *Status) Render() string {
	s.Refresh()
	return fmt.Sprintf("%s: %s", s.label, s.color)
}

func (s *Status) GetStatusLine() string {
	s.Refresh()
	st := s.engine.GetStatus()
	if st == nil {
		return "No license"
	}
	if st.Valid {
		if st.TrialActive {
			return fmt.Sprintf("Trial: %dd", st.DaysRemaining)
		}
		return fmt.Sprintf("Licensed: %dd", st.DaysRemaining)
	}
	if st.Message != "" {
		return st.Message
	}
	return "No license"
}

func (s *Status) Refresh() {
	st := s.engine.GetStatus()
	if st == nil {
		st = s.engine.Initialize()
	}
	if st != nil && st.Valid {
		if st.TrialActive {
			s.label = fmt.Sprintf("Trial: %dd", st.DaysRemaining)
			s.color = "yellow"
		} else {
			s.label = fmt.Sprintf("Licensed: %dd", st.DaysRemaining)
			s.color = "green"
		}
	} else {
		if st != nil && st.Message != "" {
			s.label = st.Message
		} else {
			s.label = "No license"
		}
		s.color = "red"
	}
}

func (s *Status) IsActive() bool {
	s.Refresh()
	return s.color == "green" || s.color == "yellow"
}

File: app/internal/publisher/template/rust/src/widgets/status_widget.rs
use crate::license_engine::LicenseEngine;

pub struct StatusWidget {
    engine: *mut LicenseEngine,
    label: String,
    active: bool,
}

impl StatusWidget {
    pub fn new(engine: &mut LicenseEngine) -> Self {
        Self {
            engine: engine as *mut LicenseEngine,
            label: String::new(),
            active: false,
        }
    }

    pub fn get_status_line(&mut self) -> String {
        self.refresh();
        let engine = unsafe { &mut *self.engine };
        if let Some(ref s) = engine.get_status_cloned() {
            if s.valid {
                if s.trial_active {
                    format!("Trial: {}d", s.days_remaining)
                } else {
                    format!("Licensed: {}d", s.days_remaining)
                }
            } else {
                s.message.clone().unwrap_or_else(|| "No license".to_string())
            }
        } else {
            "No license".to_string()
        }
    }

    pub fn refresh(&mut self) {
        let engine = unsafe { &mut *self.engine };
        let s = engine.get_status_cloned().unwrap_or_else(|| engine.initialize());
        if s.valid {
            if s.trial_active {
                self.label = format!("Trial: {}d", s.days_remaining);
            } else {
                self.label = format!("Licensed: {}d", s.days_remaining);
            }
            self.active = true;
        } else {
            self.label = s.message.unwrap_or_else(|| "No license".to_string());
            self.active = false;
        }
    }

    pub fn is_active(&self) -> bool {
        self.active
    }

    pub fn get_label(&self) -> &str {
        &self.label
    }
}
